### PR TITLE
Always cleanup the temp dirs

### LINKED ISSUE #370: Windows tests failing
The Windows tests have been failing since https://github.com/pulumi/templates/actions/runs/3039081499/jobs/4893593866.

A subset of the AWS tests fail on Windows with this:

```
  Diagnostics:
    aws:s3:Bucket (my-bucket):
      error: unable to validate AWS credentials. Make sure you have:
      
       	 • Set your AWS region, e.g. `pulumi config set aws:region us-west-2`
       	 • Configured your AWS credentials as per https://pulumi.io/install/aws.html
       	 You can also set these via cli using `aws configure`.
```

Oddly, many of the AWS tests (in fact, the ones added in the that first saw these failures) do succeed.

### PR DESCRIPTION
**Pull request by t0yv0**
Likely the CI out of disk space issues occur because we don't clean up the temp folders. https://github.com/pulumi/pulumi/issues/9437 seems to be closed now. Should it be safe to re-enable cleaning the temp folders on CI?

Fixes #370 also

### CODE REVIEW

File: tests/internal/testutils/environments.go [hunk 1]
```diff
@@ -22,10 +22,6 @@ func NewEnvironment(t *testing.T, cfg TemplateTestConfig) *ptesting.Environment
 // deleteIfNotFailed deletes the files in the testing environment if the testcase has
 // not failed. (Otherwise they are left to aid debugging.)
 func deleteIfNotFailed(e *ptesting.Environment, cfg TemplateTestConfig) {
-	if cfg.CI {
```
Comment (t0yv0) [inline tests/internal/testutils/environments.go:25]: cfg.CI was set from "CI" env which is set on GitHub actions.

**Comment by t0yv0** [comment]
I've found this was introduced in https://github.com/pulumi/templates/pull/263

**Comment by t0yv0** [comment]
@AaronFriel do we still need the workaround? I can reintroduce it scoped to Windows only. Windows runners seem to have more disk space. However Ubuntu runners are now failing due to out of disk space issues because of this workaround.

**Comment by Frassle** [comment]
windows tests don't seem happy, but the change looks like it _should_ be fine...

**Comment by t0yv0** [comment]
That's interesting. The first time around windows tests simply died (which is indeed suspect that it has something to do with this change). But the second time around (attempt #2) they reported issues:

```
unable to validate AWS credentials.
```

I'm guessing because these now take too long for AWS credentials - the creds expire. This is one hour:

```
          role-duration-seconds: 3600
```

I can extend that.

**Comment by t0yv0** [comment]
Looks like Windows tests die without output on master also https://github.com/pulumi/templates/actions/runs/3137502256/jobs/5095745367

**Comment by t0yv0** [comment]
The latest windows test failure is not related, and is in fact https://github.com/pulumi/templates/issues/386

I recommend we merge this, thoughts?

**Review by Frassle — APPROVED** [review]

**Review by cnunciato — APPROVED** [review]

**Pull request merged**